torchvision/datasets/kitti.py

```diff
@@ -0,0 +1,166 @@
+    ]
+    image_dir_name = "image_2"
+    labels_dir_name = "label_2"
+
+    def __init__(
+        self,
+        root: str,
+        split: str = None,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+        transforms: Optional[Callable] = None,
+        download: bool = False,
+    ):
+        super().__init__(
+            root,
+            transform=transform,
+            target_transform=target_transform,
+            transforms=transforms,
+        )
+        self.images = []
+        self.targets = []
+        self.root = root
+        self.split = split
+
+        if download:
+            self.download()
+        if not self._check_exists():
+            raise RuntimeError(
+                "Dataset not found. You may use download=True to download it."
+            )
+
+        location = "testing" if self.split == "test" else "training"
+        image_dir = os.path.join(self.raw_folder, location, self.image_dir_name)
+        if location == "training":
+            labels_dir = os.path.join(self.raw_folder, location, self.labels_dir_name)
+        for img_file in os.listdir(image_dir):
+            self.images.append(os.path.join(image_dir, img_file))
+            if location == "training":
+                self.targets.append(
+                    os.path.join(labels_dir, f"{img_file.split('.')[0]}.txt")
+                )
+
+    def __getitem__(self, index: int) -> Tuple[Any, Any]:
+        """Get item at a given index.
+        Args:
+            index (int): Index
+        Returns:
+            tuple: (image, target), where
+            target is a dictionary with the following keys:
+                type: Int64Tensor[N]
+                truncated: FloatTensor[N]
+                occluded: Int64Tensor[N]
+                alpha: FloatTensor[N]
+                bbox: FloatTensor[N, 4]
+                dimensions: FloatTensor[N, 3]
+                locations: FloatTensor[N, 3]
+                rotation_y: FloatTensor[N]
+                score: FloatTensor[N]
+        """
+        image = Image.open(self.images[index])
+        target = None if self.split == "test" else self._parse_target(index)
+        if self.transforms:
+            image, target = self.transforms(image, target)
+        return image, target
+
+    def _parse_target(self, index: int) -> Dict[str, Any]:
+        target: Dict[str, Any] = defaultdict(list)
+        with open(self.targets[index]) as inp:
+            content = csv.reader(inp, delimiter=" ")
+            for line in content:
+                target["type"].append(line[0])
+                target["truncated"].append(line[1])
+                target["occluded"].append(line[2])
+                target["alpha"].append(line[3])
+                target["bbox"].append(line[4:8])
+                target["dimensions"].append(line[8:11])
+                target["location"].append(line[11:14])
+                target["rotation_y"].append(line[14])
```

Comment:
Quick question: does this for loop represent different objects?
If yes, I wonder which representation might make more sense to expose to the user by default (`List[Dict]` or `Dict[List]`). Do you have any thoughts on this?

Also, just wondering if `csv` automatically casts the data in `line` to be `int` / `float` / etc, or if the data is always represented as strings?

Reply:
Yeah, this loop corresponds to different objects present in the image. That's a fair point, a list of dictionaries make more sense here.
`csv` leaves everything as strings. I think we should convert it into the right format. I'll update the code. Thanks!